File: java/filament/src/main/java/io/github/erkko68/filament/EntityManager.java
package io.github.erkko68.filament;

public class EntityManager {
    private static final EntityManager sInstance = new EntityManager();
    private long mNativeObject;

    private EntityManager() {
        mNativeObject = nGetEntityManager();
    }

    public static EntityManager get() {
        return sInstance;
    }

    public int create() {
        return nCreate(mNativeObject);
    }

    public void destroy(int entity) {
        nDestroy(mNativeObject, entity);
    }

    private static native long nGetEntityManager();
    private static native int nCreate(long nativeEntityManager);
    private static native void nDestroy(long nativeEntityManager, int entity);
}

File: kotlin/filament/src/commonMain/kotlin/io/github/erkko68/filament/EntityManager.kt
package io.github.erkko68.filament

expect class EntityManager {
    companion object {
        fun get(): EntityManager
    }

    fun create(): Entity
    fun create(n: Int): IntArray
    fun create(entities: IntArray): IntArray
    
    fun destroy(entity: Entity)
    fun destroy(entities: IntArray)
    
    fun isAlive(entity: Entity): Boolean
}
